src/obelisk_client.cpp

```diff
@@ -124,19 +126,89 @@ void obelisk_client::wait()
     clear(error::channel_timeout);
 }
 
-// Used by watch-* commands, fires registered update or unknown handlers.
+bool obelisk_client::subscribe_block(const config::endpoint& address,
+    block_update_handler on_update)
+{
+    const auto host_address = address.to_string();
+    if (block_socket_.connect(host_address) == error::success)
+    {
+        on_block_update_ = on_update;
+        return true;
+    }
+
+    return false;
+}
+
+bool obelisk_client::subscribe_transaction(
+    const config::endpoint& address, transaction_update_handler on_update)
+{
+    const auto host_address = address.to_string();
+    if (transaction_socket_.connect(host_address) == error::success)
+    {
+        on_transaction_update_ = on_update;
+        return true;
+    }
+
+    return false;
+}
+
+// Used by watch-* and subscribe-* commands, fires registered update
+// or unknown handlers.
 void obelisk_client::monitor(uint32_t timeout_seconds)
 {
     const auto deadline = asio::steady_clock::now() +
         asio::seconds(timeout_seconds);
 
     zmq::poller poller;
     poller.add(socket_);
+    poller.add(block_socket_);
+    poller.add(transaction_socket_);
     auto delay = remaining(deadline);
 
-    while (poller.wait(delay).contains(socket_.id()))
+    while (delay)
```

Comment:
boolean test on a number?

Reply:
This changes the termination behavior from the previous implementation.

Reply:
Other than the check for delay > 0 (instead of delay != 0), adding the other sockets does change things a little, in that if no identifier is found it would not terminate early.  That could still happen with a series of if/else if/else instead of how it is now.  Not sure that's better though, or could actually happen given the constrained poller set?

Reply:
Yeah, this will now consume the full deadline even if an unmonitored (or no) id is received. Not sure where this would have an impact.